sparql/SPARQLParser.g4

```diff
@@ -0,0 +1,694 @@
+ | CONTAINS OPAR expression COMMA expression CPAR
+ | STRSTARTS OPAR expression COMMA expression CPAR
+ | STRENDS OPAR expression COMMA expression CPAR
+ | STRBEFORE OPAR expression COMMA expression CPAR
+ | STRAFTER OPAR expression COMMA expression CPAR
+ | YEAR OPAR expression CPAR
+ | MONTH OPAR expression CPAR
+ | DAY OPAR expression CPAR
+ | HOURS OPAR expression CPAR
+ | MINUTES OPAR expression CPAR
+ | SECONDS OPAR expression CPAR
+ | TIMEZONE OPAR expression CPAR
+ | TZ OPAR expression CPAR
+ | NOW NIL
+ | UUID NIL
+ | STRUUID NIL
+ | MD5 OPAR expression CPAR
+ | SHA1 OPAR expression CPAR
+ | SHA256 OPAR expression CPAR
+ | SHA384 OPAR expression CPAR
+ | SHA512 OPAR expression CPAR
+ | COALESCE expressionList
+ | IF OPAR expression COMMA expression COMMA expression CPAR
+ | STRLANG OPAR expression COMMA expression CPAR
+ | STRDT OPAR expression COMMA expression CPAR
+ | SameTerm OPAR expression COMMA expression CPAR
+ | IsIRI OPAR expression CPAR
+ | IsURI OPAR expression CPAR
+ | IsBLANK OPAR expression CPAR
+ | IsLITERAL OPAR expression CPAR
+ | IsNUMERIC OPAR expression CPAR
+ | regexExpression
+ | existsFunc
+ | notExistsFunc
+ ;
+
+/// [122]    RegexExpression : 'REGEX' '(' Expression ',' Expression ( ',' Expression )? ')'
+regexExpression : REGEX OPAR expression COMMA expression ( COMMA expression )? CPAR;
+
+/// [123]    SubstringExpression : 'SUBSTR' '(' Expression ',' Expression ( ',' Expression )? ')'
+substringExpression : SUBSTR OPAR expression COMMA expression ( COMMA expression )? CPAR;
+
+/// [124]    StrReplaceExpression : 'REPLACE' '(' Expression ',' Expression ',' Expression ( ',' Expression )? ')'
+strReplaceExpression : REPLACE OPAR expression COMMA expression COMMA expression ( COMMA expression )? CPAR;
+
+/// [125]    ExistsFunc : 'EXISTS' GroupGraphPattern
+existsFunc : EXISTS groupGraphPattern;
+
+/// [126]    NotExistsFunc : 'NOT' 'EXISTS' GroupGraphPattern
+notExistsFunc : NOT EXISTS groupGraphPattern;
+
+/// [127]    Aggregate
+///           : 'COUNT' '(' 'DISTINCT'? ( '*' | Expression ) ')'
+///           | 'SUM' '(' 'DISTINCT'? Expression ')'
+///           | 'MIN' '(' 'DISTINCT'? Expression ')'
+///           | 'MAX' '(' 'DISTINCT'? Expression ')'
+///           | 'AVG' '(' 'DISTINCT'? Expression ')'
+///           | 'SAMPLE' '(' 'DISTINCT'? Expression ')'
+///           | 'GROUP_CONCAT' '(' 'DISTINCT'? Expression ( ';' 'SEPARATOR' '=' String )? ')'
+aggregate
+ : COUNT OPAR DISTINCT? ( MUL | expression ) CPAR
+ | SUM OPAR DISTINCT? expression CPAR
+ | MIN OPAR DISTINCT? expression CPAR
+ | MAX OPAR DISTINCT? expression CPAR
+ | AVG OPAR DISTINCT? expression CPAR
+ | SAMPLE OPAR DISTINCT? expression CPAR
+ | GROUP_CONCAT OPAR DISTINCT? expression ( SCOL SEPARATOR EQ string )? CPAR
+ ;
+
+/// [128]    iriOrFunction : iri ArgList?
+iriOrFunction : iri argList?;
+
+/// [129]    RDFLiteral : String ( LANGTAG | ( '^^' iri ) )?
+rdfLiteral : string ( LANGTAG | ( CARET2 iri ) )?;
```

Comment:
```suggestion
rdfLiteral : string ( LANGTAG | CARET2 iri )?;
```

Reply:
I am using the grammar myself, and do not want to change the structure of the resulting parse tree, and I also do not want to maintain 2 grammars, or update the grammar in this repo.

Reply:
Sorry, but parentheses removing doesn't change parse tree. It just makes grammar a bit more clear.

Reply:
I didn't look at all the exact proposed changes, so that might be the case. Regardless, I like the grammar to look like the way the spec was written, so I also don't want to make changes (besides the comments). No offense, but I'll opt-out this contribution.